### PR TITLE
CLOUD-3121 - EAP 7.2 / OpenJDK 11 image support

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.jboss.org/browse/CLOUD-3121
Signed-off-by: Ken Wills <kwills@redhat.com>

### CODE REVIEW

File: os-eap-launch/added/launch/datasource-common.sh [hunk 1]
```diff
@@ -494,8 +494,13 @@ function inject_datasource() {
...
   url=$(find_env "${prefix}_URL")
 
-  case "$db" in
+  IFS=. read -r -a java_version <<< "${JAVA_VERSION}"
+  java_maj_version=${java_version[0]:-8}
+  # JDK >= 11 doesn't have the default drivers present.
+  if [ ${java_maj_version} -lt 11 ]; then
+    case "$db" in
     "MYSQL")
+      log_warning "Datasource ${service_name} is using the bundled MySQL driver. This driver has been deprecated and will be dropped in a future release. Please see the product documentation for more information on installing database drivers."
```
Comment (bstansberry) [inline os-eap-launch/added/launch/datasource-common.sh]: https://issues.jboss.org/browse/JBEAP-16626 is triaged for 7.2.2, not 7.2.1, so let's not do this yet.

Also, a commit for it should reference CLOUD-3143.

Comment (luck3y) [reply os-eap-launch/added/launch/datasource-common.sh]: OK, this change is dropped. I'll reply to your other comment below.

**Comment by luck3y** [comment]
@bstansberry Thats a good point, we should create a new module that drops the default behavior and EAP can swtich to that. The legacy module can remain unmodified for others to use.

**Comment by luck3y** [comment]
@bstansberry This is ready for a look, I'm doing a quick test in OS to make sure everything is OK, then I'll build an image to test with.

**Comment by luck3y** [comment]
Note that https://github.com/jboss-container-images/jboss-eap-7-openshift-image/pull/218 has an additional change that uses the new module. [OT, not sure why it didn't reflag that one for review....]

File: os-eap-launch/added/launch/datasource-common.sh [hunk 2]
```diff
@@ -494,7 +494,11 @@ function inject_datasource() {
 
   url=$(find_env "${prefix}_URL")
 
-  case "$db" in
```
Comment (bstansberry) [inline os-eap-launch/added/launch/datasource-common.sh]: I don't think this file should have any changes.

File: os-eap7-launch/tests/bats/datasource/configure.bats [hunk 3]
```diff
@@ -290,7 +290,10 @@ load common
     run inject_datasources
 
     expected="WARN Missing configuration for XA datasource TEST. Either TEST_XA_CONNECTION_PROPERTY_URL or TEST_XA_CONNECTION_PROPERTY_ServerName, and TEST_XA_CONNECTION_PROPERTY_PortNumber, and TEST_XA_CONNECTION_PROPERTY_DatabaseName is required. Datasource will not be configured."
-    [ "$output" = "$expected" ]
```
Comment (bstansberry) [inline os-eap7-launch/tests/bats/datasource/configure.bats]: I don't these changes apply.

File: os-eap-launch/added/launch/datasource-common.sh (hunk 2 above)
Comment (luck3y) [reply os-eap-launch/added/launch/datasource-common.sh]: Sorry, my error, I thought i'd revert those, will update shortly.

**Pull request merged**